### ISSUE TITLE
Remember Discord Session/Login

### ISSUE THREAD

**Issue by GNEUGNEUGNEU** [zorn-v/nextcloud-social-login#353]
Hi, i'm using Discord oauth and everytime i want to go to my cloud i have to log and approve with discord that i want to use my discord account for nextcloud, how i can set up to always be connected ? Thanks

**Comment by zorn-v** [comment]
It is discord related. They ask confirm on every login (what it right by the way).

**Comment by GNEUGNEUGNEU** [comment]
yes, but is it possible to just stay logged in nextcloud ? i tryed to modify the config.php about cookie session lifetime but it doesnt change anything.

**Comment by zorn-v** [comment]
> yes, but is it possible to just stay logged in nextcloud ?

Nope. OAuth don't work like this.

**Comment by zorn-v** [comment]
Possibly I misunderstand you. You should be logged in nextcloud anyway.
Can you reproduce login flow that "incorrect" on your mean ?
Like video capture with "peek", or some what ?

**Comment by GNEUGNEUGNEU** [comment]
if i leave my navigator open, it's all good, i don't have to reconnect, but if i close my navigator and relaunch it i will be disconnected from my nextcloud server and i'll have to reconnect, it happen on different browsers/pc/users

**Comment by zorn-v** [comment]
![изображение](https://user-images.githubusercontent.com/12619075/170020728-5832bede-57ac-4c39-bd92-2bc83e6b0337.png)
Does not have such problems with this option. Restore previews tabs in english

**Comment by GNEUGNEUGNEU** [comment]
i use opera and it also restores all my tabs, and still i have to reconnect, maybe there is some log i can share ?

**Comment by zorn-v** [comment]
F12 and so on.
Your issue definitely not related to this plugin.

**Comment by GNEUGNEUGNEU** [comment]
i have an admin user that do not use this plugin and i can stay logged in for days and close and reopen browser and i'll still be logged

**Comment by GNEUGNEUGNEU** [comment]
maybe it's just with discord or some setting i misconfigured

**Comment by zorn-v** [comment]
> maybe it's just with discord or some setting i misconfigured

THIS. You configured al right. But as I can "imagine", your users see "confirm that app" in discord ?

**Comment by WhoAmI0501** [comment]
Hi,
i am facing a similiar problem with custom OAuth2 via Mailcow at a community project.

I searched in some older Issues and found this problem description: https://github.com/zorn-v/nextcloud-social-login/issues/192#issuecomment-757062677

I reviewed the cookies on two of my servers and have had seen, that the exactly same behaviour is showing up again. The two mentioned cookies are missing again on the instance with login via OAuth2.

Both Nextcloud instances where i tested it were on version 24.0.1 and the one with OAuth2 authentication used the Social Login App version 4.15.2.

Maybe this helps to find the problem. @GNEUGNEUGNEU May you want to take a look, whether the mentioned cookies are also missing on your Nextcloud instance?

**Comment by zorn-v** [comment]
Check v4.15.3

**Comment by GNEUGNEUGNEU** [comment]
a bit late, but you're a boss
